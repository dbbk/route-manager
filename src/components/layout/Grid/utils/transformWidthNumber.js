export default function transformWidthNumber(w: number) {
  return w >= 0 && w <= 1 ? `${w * 100}%` : `${w}px`;
}
