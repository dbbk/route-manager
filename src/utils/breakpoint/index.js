/* eslint consistent-return: 0 default-case: 0 */
/* stylelint-disable */

import { css } from 'styled-components';
import theme from '../theme';

export default function breakpoint(label: string) {
  return (...args: any) => () => {
    switch (label) {
      case 'xs':
        return css`
          @media (max-width: ${theme.breakpoints.xs}px) {
            ${css(...args)};
          }
        `;

      case 'sm':
        return css`
          @media (min-width: ${theme.breakpoints.xs}px) and (max-width: ${theme
              .breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'md':
        return css`
          @media (min-width: ${theme.breakpoints.sm}px) and (max-width: ${theme
              .breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'lg':
        return css`
          @media (min-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'to-sm':
        return css`
          @media (max-width: ${theme.breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'to-md':
        return css`
          @media (max-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'to-lg':
        return css`
          @media (max-width: ${theme.breakpoints.lg}px) {
            ${css(...args)};
          }
        `;

      case 'from-xs':
        return css`
          @media (min-width: ${theme.breakpoints.xs}px) {
            ${css(...args)};
          }
        `;

      case 'from-sm':
        return css`
          @media (min-width: ${theme.breakpoints.sm}px) {
            ${css(...args)};
          }
        `;

      case 'from-md':
        return css`
          @media (min-width: ${theme.breakpoints.md}px) {
            ${css(...args)};
          }
        `;

      case 'from-lg':
        return css`
          @media (min-width: ${theme.breakpoints.lg}px) {
            ${css(...args)};
          }
        `;

      case 'from-xlg':
        return css`
          @media (min-width: ${theme.breakpoints.xlg}px) {
            ${css(...args)};
          }
        `;
    }
  };
}
