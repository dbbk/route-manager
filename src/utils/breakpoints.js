// @flow

// Breakpoints are defined as a map of (name: minimum width)
// https://zpl.io/25RJPjV
const breakpoints = {
  xs: 0, // officially we support a minimum width of 320px
  sm: 569,
  md: 769,
  lg: 961,
  xlg: 1221, // DEPRECATED
};

export default breakpoints;
