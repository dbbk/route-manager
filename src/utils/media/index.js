// @flow

import bp from 'src/utils/breakpoints';

export default function media(label: string) {
  return () => {
    switch (label) {
      case 'xs':
        return `@media (max-width: ${bp.sm - 1}px)`;

      case 'sm':
        return `@media (min-width: ${bp.sm}px) and (max-width: ${bp.md - 1}px)`;

      case 'md':
        return `@media (min-width: ${bp.md}px) and (max-width: ${bp.lg - 1}px)`;

      case 'lg':
        return `@media (min-width: ${bp.lg}px)`;

      case 'to-sm':
        return `@media (max-width: ${bp.sm - 1}px)`;

      case 'to-md':
        return `@media (max-width: ${bp.md - 1}px)`;

      case 'to-lg':
        return `@media (max-width: ${bp.lg - 1}px)`;

      case 'from-sm':
        return `@media (min-width: ${bp.sm}px)`;

      case 'from-md':
        return `@media (min-width: ${bp.md}px)`;

      case 'from-lg':
        return `@media (min-width: ${bp.lg}px)`;

      default:
        return null;
    }
  };
}
