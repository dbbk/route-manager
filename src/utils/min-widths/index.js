// @flow

// minWidths defines the minimum widths of our responsive breakpoints
// as a map of (name: minimum width)
// https://zpl.io/25RJPjV
const minWidths = {
  xs: 0, // officially we support a minimum width of 320px
  sm: 569,
  md: 769,
  lg: 961,
  xlg: 1221, // DEPRECATED
};

export default minWidths;
