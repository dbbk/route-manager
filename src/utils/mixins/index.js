// @flow

import { css } from 'styled-components';
import { prop, switchProp, withProp } from 'styled-tools';
import { darken, lighten } from 'polished';

export const colorMixin = switchProp('color', {
  black: css`
    ${prop('theme.colors.greyscale.black')};
  `,
  grey: css`
    ${prop('theme.colors.greyscale.dark')};
  `,
  white: css`
    ${prop('theme.colors.white')};
  `,
  red: css`
    ${prop('theme.colors.danger')};
  `,
  primary: css`
    ${prop('theme.colors.brand.primary')};
  `,
  inherit: 'inherit',
});

// NOTE while we are lighening and darkening colours using withProp,
// "babel-plugin-polished" will not work.
export const hoverColorMixin = switchProp('color', {
  black: css`
    ${withProp('theme.colors.greyscale.black', lighten(0.1))};
  `,
  grey: css`
    ${withProp('theme.colors.greyscale.dark', darken(0.1))};
  `,
  white: css`
    ${withProp('theme.colors.white', darken(0.1))};
  `,
  red: css`
    ${withProp('theme.colors.danger', darken(0.1))};
  `,
  primary: css`
    ${withProp('theme.colors.brand.primary', darken(0.1))};
  `,
  inherit: 'inherit',
});
