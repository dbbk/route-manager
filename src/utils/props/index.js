// @flow

export function blacklistProps(props: {}, blacklist: Array<string> = []): {} {
  const newProps = { ...props };

  blacklist.forEach(blacklistedProp => delete newProps[blacklistedProp]);

  return newProps;
}
