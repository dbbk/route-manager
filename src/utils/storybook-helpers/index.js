// @flow
import styled from 'styled-components';

const BlackBackground = styled.div`
  background-color: black;
  padding: 1em;
`;

const WhiteBackground = styled.div`
  background-color: white;
  padding: 1em;
`;

const GreyBackground = styled.div`
  background-color: lightgrey;
  padding: 1em;
  text-align: center;
  font-family: monospace;
`;

const FullWidthBackground = styled.div`
  height: 100vh;
  width: 100vw;
`;

export const Background = {
  White: WhiteBackground,
  Black: BlackBackground,
  Grey: GreyBackground,
  FullWidth: FullWidthBackground,
};
