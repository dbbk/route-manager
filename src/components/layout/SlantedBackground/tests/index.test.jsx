// @flow

import React from 'react';
import { SlantedBackground } from '@festicket/react-ui-components';
import { renderWithTheme } from 'src/tests/utils';

describe('<SlantedBackground />', () => {
  test('with initial props', () => {
    const tree = renderWithTheme(
      <SlantedBackground>
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with primary slantVariant', () => {
    const tree = renderWithTheme(
      <SlantedBackground slantVariant="primary">
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with grey slantVariant', () => {
    const tree = renderWithTheme(
      <SlantedBackground slantVariant="grey">
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with primary backgroundVariant and white slantVariant', () => {
    const tree = renderWithTheme(
      <SlantedBackground backgroundVariant="primary" slantVariant="white">
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with primary backgroundVariant and white slantVariant and slantVerticalOffset', () => {
    const tree = renderWithTheme(
      <SlantedBackground
        backgroundVariant="primary"
        slantVariant="white"
        slantVerticalOffset
      >
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with primary backgroundVariant and white slantVariant and contentVerticalOffset', () => {
    const tree = renderWithTheme(
      <SlantedBackground
        backgroundVariant="primary"
        slantVariant="white"
        contentVerticalOffset
      >
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('with primary backgroundVariant and white slantVariant and slantVerticalOffset and to-left slantHorizontalDirection', () => {
    const tree = renderWithTheme(
      <SlantedBackground
        backgroundVariant="primary"
        slantVariant="white"
        slantVerticalOffset
        slantHorizontalDirection="to-left"
      >
        <span>Content</span>
      </SlantedBackground>,
    );
    expect(tree).toMatchSnapshot();
  });
});
