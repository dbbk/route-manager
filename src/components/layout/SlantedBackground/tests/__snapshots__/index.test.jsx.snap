// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlantedBackground /> with grey slantVariant 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 200px;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  background: #F5F5F5;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
  clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;

exports[`<SlantedBackground /> with initial props 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 200px;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
  clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;

exports[`<SlantedBackground /> with primary backgroundVariant and white slantVariant 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 200px;
  background: #8C64CB;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  background: white;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
  clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;

exports[`<SlantedBackground /> with primary backgroundVariant and white slantVariant and contentVerticalOffset 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 200px;
  background: #8C64CB;
  padding-top: 200px;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  background: white;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
  clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;

exports[`<SlantedBackground /> with primary backgroundVariant and white slantVariant and slantVerticalOffset 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 400px;
  background: #8C64CB;
}

.c0::after {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 200px;
  background: white;
  z-index: -1;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 200px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  background: white;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
  clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;

exports[`<SlantedBackground /> with primary backgroundVariant and white slantVariant and slantVerticalOffset and to-left slantHorizontalDirection 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 400px;
  background: #8C64CB;
}

.c0::after {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 200px;
  background: white;
  z-index: -1;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 200px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  background: white;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 0,0 100% );
  clip-path: polygon( 0 0,100% 0,100% 0,0 100% );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;

exports[`<SlantedBackground /> with primary slantVariant 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  z-index: 0;
  min-height: 200px;
}

.c0::before {
  display: block;
  content: '';
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 2000px;
  width: 100%;
  z-index: -1;
  height: 200px;
  background: #8C64CB;
  -webkit-clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
  clip-path: polygon( 0 0,100% 0,100% 100%,0 0 );
}

<div
  className="c0"
>
  <span>
    Content
  </span>
</div>
`;
